R/render.R

```diff
@@ -15,45 +18,76 @@ in_age_range <- function(birth, timestep, lower, upper) {
 #' 
 #' @noRd
 create_prevelance_renderer <- function(
-  state,
-  birth,
-  immunity,
-  parameters,
-  renderer
-  ) {
+    state,
+    birth,
+    immunity,
+    parameters,
+    renderer
+) {
   function(timestep) {
+    
     asymptomatic <- state$get_index_of('A')
-    prob <- probability_of_detection(
-      get_age(birth$get_values(asymptomatic), timestep),
-      immunity$get_values(asymptomatic),
-      parameters
-    )
-    asymptomatic_detected <- bitset_at(asymptomatic, bernoulli_multi_p(prob))
-
+    
+    if(parameters$parasite == "falciparum"){
+      prob <- probability_of_detection(
+        get_age(birth$get_values(asymptomatic), timestep),
+        immunity$get_values(asymptomatic),
+        parameters
+      )
+      asymptomatic_detected <- bitset_at(asymptomatic, bernoulli_multi_p(prob))
+    } else {
+      asymptomatic_detected <- asymptomatic
+    }
+    
     clinically_detected <- state$get_index_of(c('Tr', 'D'))
-    detected <- clinically_detected$copy()$or(asymptomatic_detected)
-
+    lm_detected <- detected <- clinically_detected$copy()$or(asymptomatic_detected)
+    pcr_detected <- state$get_index_of(c('Tr', 'D', 'A', 'U'))
+    
     for (i in seq_along(parameters$prevalence_rendering_min_ages)) {
       lower <- parameters$prevalence_rendering_min_ages[[i]]
       upper <- parameters$prevalence_rendering_max_ages[[i]]
       in_age <- in_age_range(birth, timestep, lower, upper)
+      
+      # render age
       renderer$render(
         paste0('n_', lower, '_', upper),
         in_age$size(),
         timestep
       ) 
+      
+      # render pcr detection
       renderer$render(
-        paste0('n_detect_', lower, '_', upper),
-        in_age$copy()$and(detected)$size(),
-        timestep
-      )
-      renderer$render(
-        paste0('p_detect_', lower, '_', upper),
-        in_age$copy()$and(clinically_detected)$size() + sum(
-          prob[bitset_index(asymptomatic, in_age)]
-        ),
+        paste0('n_detect_pcr_', lower, '_', upper),
+        in_age$copy()$and(pcr_detected)$size(),
         timestep
       )
+      
+      if(parameters$parasite == "falciparum"){
+        # render lm detection (falciparum): integer sample
+        renderer$render(
+          paste0('n_detect_lm_', lower, '_', upper),
+          in_age$copy()$and(lm_detected)$size(),
+          timestep
+        )
+        
+        # render lm detection (falciparum): summed probability
+        renderer$render(
+          paste0('p_detect_lm_', lower, '_', upper),
+          in_age$copy()$and(clinically_detected)$size() + sum(
+            prob[bitset_index(asymptomatic, in_age)]
+          ),
+          timestep
+        )
+        
+      } else if (parameters$parasite == "vivax") {
+        
+        # render lm detection (vivax)
+        renderer$render(
+          paste0('n_detect_lm_', lower, '_', upper),
+          in_age$copy()$and(lm_detected)$size(),
+          timestep
+        )
+      }
```

Comment:
Sure thing - I think I'd been separating them because the calculations were different, but I've since included an extra if statement earlier that does that calculation at an earlier point.